flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManagerTestBase.java

```diff
@@ -115,7 +114,8 @@ static TaskExecutorConnection createTaskExecutorConnection() {
         private final TaskManagerTracker taskManagerTracker = new FineGrainedTaskManagerTracker();
         private final SlotStatusSyncer slotStatusSyncer =
                 new DefaultSlotStatusSyncer(Time.seconds(10L));
-        private final ScheduledExecutor scheduledExecutor = TestingUtils.defaultScheduledExecutor();
+        private final ManuallyTriggeredScheduledExecutor scheduledExecutor =
+                new ManuallyTriggeredScheduledExecutor();
```

Comment:
Not sure if we need to manually trigger the scheduled events for all the tests.
I think we can keep the default `TestingUtils.defaultScheduledExecutor()`, and allows the test cases to provide a `ManuallyTriggeredScheduledExecutor` when it's needed.